mailet/icalendar/src/main/java/org/apache/james/transport/mailets/ICalendarParser.java

```diff
@@ -0,0 +1,128 @@
+import net.fortuna.ical4j.data.ParserException;
+import net.fortuna.ical4j.model.Calendar;
+
+/**
+ * <p>
+ * This mailet can be combined with the Strip attachment mailet.
+ * </p>
+ * <p>
+ * The ICS body part byte array arranged as map then this mailet should look for ICS and parse it with Ical4J then store it as a mail attribute
+ * </p>
+ * <p>
+ * Configuration: The mailet contains 2 mandatory attributes
+ * </p>
+ * <p>
+ *
+ * <pre>
+ *   &lt;mailet match=&quot;All&quot; class=&quot;ICalendarParser&quot; &gt;
+ *     &lt;sourceAttribute&gt;source.attribute.name&lt;/sourceAttribute&gt;  &lt;!-- The attribute which contains output value of StripAttachment mailet -- &gt;
+ *     &lt;destAttribute&gt;dest.attribute.name&lt;/destAttribute&gt;  &lt;!-- The attribute store the map of Calendar -- &gt;
+ *   &lt;/mailet &gt;
+ *
+ * </pre>
+ *
+ * </p>
+ */
+public class ICalendarParser extends GenericMailet {
+    public static final String SOURCE_ATTRIBUTE_PARAMETER_NAME = "sourceAttribute";
+    public static final String DESTINATION_ATTRIBUTE_PARAMETER_NAME = "destinationAttribute";
+
+    public static final String SOURCE_ATTRIBUTE_PARAMETER_DEFAULT_VALUE = "icsAttachments";
+    public static final String DESTINATION_ATTRIBUTE_PARAMETER_DEFAULT_VALUE = "calendars";
+
+    private String sourceAttributeName;
+    private String destinationAttributeName;
+
+    @Override
+    public void init() throws MessagingException {
+        sourceAttributeName = getInitParameter(SOURCE_ATTRIBUTE_PARAMETER_NAME, SOURCE_ATTRIBUTE_PARAMETER_DEFAULT_VALUE);
+        if (Strings.isNullOrEmpty(sourceAttributeName)) {
+            throw new MessagingException("source attribute cannot be empty");
+        }
+        destinationAttributeName = getInitParameter(DESTINATION_ATTRIBUTE_PARAMETER_NAME, DESTINATION_ATTRIBUTE_PARAMETER_DEFAULT_VALUE);
+        if (Strings.isNullOrEmpty(destinationAttributeName)) {
+            throw new MessagingException("destination attribute cannot be empty");
+        }
+    }
+
+    @VisibleForTesting
+    public String getSourceAttributeName() {
+        return sourceAttributeName;
+    }
+
+    @VisibleForTesting
+    public String getDestinationAttributeName() {
+        return destinationAttributeName;
+    }
+
+    @Override
+    public void service(Mail mail) throws MessagingException {
+        Object icsAttachmentsObj = mail.getAttribute(sourceAttributeName);
+        if (icsAttachmentsObj == null || !(icsAttachmentsObj instanceof Map)) {
+            return;
+        }
+
+        Map<String, byte[]> icsAttachments = (Map<String, byte[]>) icsAttachmentsObj;
+        Map<String, Calendar> calendars = icsAttachments.entrySet()
+            .stream()
+            .flatMap(entry -> createCalendar(entry.getKey(), entry.getValue()))
+            .collect(Guavate.toImmutableMap(Pair::getKey, Pair::getValue));
+
+        mail.setAttribute(destinationAttributeName, (Serializable) calendars);
```

Comment:
Yes, the Calendar object is serializable and String also. So it's pretty sure that calendars are serializable.